spec/factories/order_factory_override.rb

```diff
@@ -0,0 +1,47 @@
+require 'spree/testing_support/factories/order_factory'
+
+FactoryGirl.define do
+  # The order with line items factory was updated in 1.2 to take a stock
+  # location as a transient attribute. This brings that change so we can
+  # specify the stock location which is used for the origin address for packages
+  # and can be removed when we no longer want to support Solidus 1.0 and 1.1.
+  factory :order_with_line_items_and_stock_location, parent: :order do
+    bill_address
+    ship_address
+
+    transient do
+      line_items_count 1
+      line_items_attributes { [{}] * line_items_count }
+      shipment_cost 100
+      shipping_method nil
+      stock_location { create(:stock_location) }
```

Comment:
That is an option, unfortunately it means we'll have to keep updating the original stock location with  the address we want to use before each example.